From: Victor Seva <vseva@sipwise.com>
Date: Tue, 12 Apr 2022 13:36:34 +0200
Subject: pv_headers: fix build warning

pvh_func.c:392:35: warning: division 'sizeof (char *) / sizeof (char)'
pvh_func.c:407:66: warning: division 'sizeof (char *) / sizeof (char)'
---
 src/modules/pv_headers/pvh_func.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/modules/pv_headers/pvh_func.c b/src/modules/pv_headers/pvh_func.c
index fccd077..6249768 100644
--- a/src/modules/pv_headers/pvh_func.c
+++ b/src/modules/pv_headers/pvh_func.c
@@ -385,8 +385,7 @@ int pvh_remove_header_param(struct sip_msg *msg, int idx, str *hname, str *eleme
 	char *token;
 	char *result = (char*)pkg_malloc(elements->len - toRemove->len);
 	char *t = (char*)pkg_malloc(elements->len);
-	int maxSize = sizeof(elements->s)/sizeof(char);
-	maxSize = maxSize < elements->len?elements->len:maxSize;
+	int maxSize = elements->len;
 
 	if (result == NULL || t == NULL)
 	{
@@ -400,7 +399,7 @@ int pvh_remove_header_param(struct sip_msg *msg, int idx, str *hname, str *eleme
 	token = strtok_r(t, ", ", &next_token);
 	while(token)
 	{
-		int notTarget = strncmp(token, toRemove->s, sizeof(toRemove->s)/sizeof(char));
+		int notTarget = strncmp(token, toRemove->s, toRemove->len);
 		if (notTarget)
 		{
 			int n = snprintf(result + offset, maxSize - offset, "%s", token);
